src/Engine/ProtoCore/Parser/AssociativeAST.cs

```diff
@@ -2803,18 +2804,45 @@ public static BooleanNode BuildBooleanNode(bool value)
         }
 
         /// <summary>
+        /// Get the default value of Type "t"
+        /// </summary>
+        /// <param name="t">The type that we want to generate a default value for</param>
+        /// <returns>The default value</returns>
+        private static object GetDefault(System.Type t)
+        {
+            Func<object> f = GetDefault<object>;
+            return f.Method.GetGenericMethodDefinition().MakeGenericMethod(t).Invoke(null, null);
+        }
+
+        private static T GetDefault<T>() => default;
+
+        /// <summary>
+        /// Builds an AssociativeNode (int, double, null etc) from the input value or type
+        /// </summary>
+        /// <param name="value">The value of the object</param>
+        /// <param name="type">The target type that the "value" parameter represents</param>
+        /// <returns>AssociativeNode</returns>
+        internal static AssociativeNode BuildPrimitiveNodeFromObjectAndType(object value, System.Type type)
+        {
+            if (value is Missing)
```

Comment:
Why do we want to import these interop types? Which nodes expose these types?

Reply:
Well ...all the Excel interop types used by the existing nodes are imported (since the types are embedded they appear as any other types in the assembly).
![image](https://github.com/DynamoDS/Dynamo/assets/46732933/816bd11c-47cb-4db6-982c-bc2d23316e0f)

Reply:
All these types do have a COM specific atribute {[System.Runtime.InteropServices.ComImportAttribute()]}
We could try to filter out COM types that are coming from DSOffice.dll

Dynamo does support embedded types. There might be valid cases where users want to import nodes from embedded types. We had this issue reported https://forum.dynamobim.com/t/fail-to-load-two-packages-access-to-the-same-api/83556/30
This was added to our docs as a response